crates/lunatic-distributed/src/control/client.rs

```diff
@@ -56,11 +56,12 @@ impl Client {
         };
         // Spawn reader task before register
         tokio::task::spawn(reader_task(client.clone()));
-        tokio::task::spawn(refresh_nodes_task(client.clone()));
         let Registered {
             node_id,
             signed_cert,
         } = client.send_registration(signing_request).await?;
+        // Refresh nodes after registration
+        tokio::task::spawn(refresh_nodes_task(client.clone()));
```

Comment:
You are spawning a task here. This means that it will be scheduled at one point in the future, but we can't guarantee when this will happen. The `register` function could finish earlier and wasm execution might start already. I would prefer to have here an explicit

```rust
client.refresh_nodes().await;
```

call. This will for sure block the progress until the list is received.

Spawning the `refresh_nodes_task` can be done earlier, like it was before.

Reply:
You are correct!